examples/lang-err-integration-tests/contract-ref/lib.rs

```diff
@@ -0,0 +1,129 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[ink::contract]
+mod contract_ref {
+    use integration_flipper::FlipperRef;
+
+    #[ink(storage)]
+    pub struct ContractRef {
+        flipper: FlipperRef,
+    }
+
+    impl ContractRef {
+        #[ink(constructor)]
+        pub fn new(version: u32, flipper_code_hash: Hash) -> Self {
+            let salt = version.to_le_bytes();
+            let flipper = FlipperRef::default()
+                .endowment(0)
+                .code_hash(flipper_code_hash)
+                .salt_bytes(salt)
+                .instantiate()
+                .unwrap_or_else(|error| {
+                    panic!("failed at instantiating the Flipper contract: {:?}", error)
+                });
+
+            Self { flipper }
+        }
+
+        #[ink(message)]
+        pub fn flip(&mut self) {
+            self.flipper.flip();
+        }
+
+        #[ink(message)]
+        pub fn flip_check(&mut self) {
+            self.flipper
+                .flip_checked()
```

Comment:
Would testing `err_flip_checked` be useful do you think? As in it will be a `Ok(Err(()))`